### DISCUSSION TITLE
Is it possible to create an "ID mixin"?

### DISCUSSION THREAD

**Discussion by GV14982** [ent/ent#3773] [Q&A]
Hi, I want to make it easier to setup a custom ID field using ulids. Is it possible to use a mixin to add that to my models rather than adding it directly to the fields of the model? I tried doing it and I was getting issues, but I'm not sure if I'm missing something.

**Comment by stavros-k** [comment]
It usually helps someone to have something to work on when asking something.
A sample of what you tried and what issues you faced is a good start.

> Not affiliated with Ent, I'm just a guy :D

**Reply by GV14982** [reply]
Fair, I guess I just wanted to ask if it was possible first before adding a bunch of extra info.

**Comment by keenanwl** [comment]
Is this what you're looking for? [link](https://github.com/ent/contrib/blob/master/entgql/internal/todopulid/ent/schema/pulid/mixin.go)

This is pretty close to what I am using.
